src/analyses/ai.h

```diff
@@ -75,6 +75,12 @@ class ai_domain_baset
   // a reasonable entry-point state
   virtual void make_entry()=0;
 
+  virtual bool is_bottom() const=0;
+
+  // the analysis doesn't use this,
+  // and domains may refuse to implement it.
```

Comment:
Yeah... I'm going to remove the comment because we're not using is_top() right now but ... it's the kind of thing we may well start using.  As there is a make_top() method the domain must have a representation of it so it shouldn't be too great a burden on the programmer.